de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,189 @@
+		configurationTree = addElement(ConfigurationTree.class, conceptPs, repositoryNavigatorItem);
+		elementConfiguration = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+				
+		createNewDiagramForTreeItem(configurationTree, DiagramType.interfaces);
+		diagramEditor = getOpenedDiagramEditorbyTitle("newDiagram");
+	}
+	
+	@Test
+	public void addInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		undoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		redoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}	
+
+	@Test
+	public void deleteObjectOutsideDiagramUpdateDiagramTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		delete(elementConfiguration);
+		updateActiveDiagram(diagramEditor);
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}
+	
+	@Test
+	public void dragDropTreeinDiagram() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(configurationTree, diagramEditor);
+		Assert.assertEquals(0, diagramEditor.selectedEditParts().size(), 0);
+	}
+	
+	@Test
+	public void addInterfaceEndDiagramElementUndoRedoTest() {
+		addElement(InterfaceEnd.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		undoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		redoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}
+	
+	@Test
+	public void dragDropAssignInterfaceTypeTest() {
+		SWTBotTreeItem interfaceTypeCollectionItem = addElement(InterfaceTypeCollection.class, conceptFuncElectrical, repositoryNavigatorItem);
+		SWTBotTreeItem interfaceTypeItem = addElement(InterfaceType.class, conceptFuncElectrical, interfaceTypeCollectionItem);
+		SWTBotTreeItem interfaceEndItem = addElement(InterfaceEnd.class, conceptFuncElectrical, elementConfiguration);
+		
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor, 0, 0);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		SWTBotGefEditPart editPart = diagramEditor.getEditPart("ElementConfiguration");
+		SWTBotGefEditPart child = editPart.children().get(0);
```

Comment:
Instead generic names, maybe we can find some more informative names. I think the child here is the diagram element for an interface end? It should be clear from the name alone.